app/views/interactive_pages/edit.html.haml

```diff
@@ -45,22 +45,27 @@
     %span.preview
       = link_to "Preview", preview_activity_page_path(@activity, @page), :target => 'new'
   %div.edit_interactive_page
-    = form_for @page, :remote => true do |f|
+    = form_for @page do |f|
       = f.select :layout, options_for_select(InteractivePage::LAYOUT_OPTIONS.map { |l| [l[:name], l[:class_val]] }, @page.layout), {},  { :onchange => 'this.form.submit();', :title => 'Page layout' }
       = f.select :embeddable_display_mode, InteractivePage::EMBEDDABLE_DISPLAY_OPTIONS, {}, { :onchange => 'this.form.submit();', :title => 'Display of embeddables: stacked for scrolling, or carousel for sequential display' }
-      %span.page-section-controls#page-section-controls     
-        %label{:id => 'page_intro',:for => "interactive_page_show_introduction"}    
+      %p.page-section-controls#page-section-controls
+        %label
           = f.check_box :show_introduction, :onchange => 'this.form.submit();' 
           Page introduction
-        %label{:id => 'info_block',:for => "interactive_page_show_info_assessment"}
+        %label
           = f.check_box :show_info_assessment, :onchange => 'this.form.submit();'
           Info/Assessment block
-        %label{:id => 'interactive_box',:for => "interactive_page_show_interactive"}
+        %label
           = f.check_box :show_interactive, :onchange => 'this.form.submit();'
           Interactive box
-        %label{:id => 'page_sidebar',:for => "interactive_page_show_sidebar"}
+        %label
           = f.check_box :show_sidebar, :onchange => 'this.form.submit();'
-          Page sidebar text  
+          Page sidebar text
+        - InteractivePage.registered_additional_sections.each do |s|
+          %label
+            = f.check_box "show_#{s[:name]}", :onchange => 'this.form.submit();'
```

Comment:
Yes, only checkboxes are using dynamic `show_` methods. Do you want me to remove them?
I assumed that they will simplify view and controller code (update method), but yes, it can be implemented without them too.